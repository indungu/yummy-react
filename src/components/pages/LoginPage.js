import React, { Component } from 'react';
import { toast, ToastContainer } from 'react-toastify';
import { Redirect } from 'react-router-dom';
import LoginForm from '../forms/LoginForm';
import axiosInstance from '../common/AxiosIntance';
import NavBar from '../common/NavBar';
import Footer from '../common/Footer';

export default class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      redirect: false,
      email: '',
      password: '',
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    console.log(this.state);
    // Get request data from form state.
    const requestData = {
      email: this.state.email,
      password: this.state.password,
    };

    // Make call to api through axios
    axiosInstance
      .post('/auth/login', requestData)
      .then((response) => {
        window.localStorage.setItem('token', response.data.access_token);
        this.props.history.push('/dashboard');
        toast.success(response.data.message);
      })
      .catch((error) => {
        if (error.response) {
          toast.error(error.response.data.message);
        }
      });
  }
  render() {
    const token = window.localStorage.getItem('token');
    if (token) {
      return <Redirect to="/dashboard" />;
    }
    return (
      <div>
        <NavBar />
        <ToastContainer />
        <div className="container">
          <div className="w-40">
            <LoginForm
              onChange={this.onChange}
              onSubmit={this.onSubmit}
            />
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
