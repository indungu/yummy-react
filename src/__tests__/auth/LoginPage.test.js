import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import LoginPage from '../../components/pages/LoginPage';

describe('Login', () => {
  const wrapper = shallow(<LoginPage />);
  it('Renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders all the divs', () => {
    expect(wrapper.find('div').length).toBe(3);
  });
});
