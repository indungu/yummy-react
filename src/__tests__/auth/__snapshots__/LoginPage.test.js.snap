// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders properly 1`] = `
<div>
  <NavBar />
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <div
    className="container"
  >
    <div
      className="w-40"
    >
      <LoginForm
        onChange={[Function]}
        onSubmit={[Function]}
      />
    </div>
  </div>
  <Footer />
</div>
`;
